modules/openxoutstreamBidAdapter.js

```diff
@@ -0,0 +1,216 @@
+      'unit': 'string',
+    }, params);
+  }
+};
+
+function getViewportDimensions(isIfr) {
+  let width;
+  let height;
+  let tWin = window;
+  let body;
+
+  if (isIfr) {
+    let tDoc;
+    try {
+      tWin = window.top;
+      tDoc = window.top.document;
+    } catch (e) {
+      return;
+    }
+    body = tDoc.body;
+
+    width = tWin.innerWidth || docEl.clientWidth || body.clientWidth;
+    height = tWin.innerHeight || docEl.clientHeight || body.clientHeight;
+  } else {
+    width = tWin.innerWidth || docEl.clientWidth;
+    height = tWin.innerHeight || docEl.clientHeight;
+  }
+
+  return `${width}x${height}`;
+}
+
+function buildCommonQueryParamsFromBids(bids, bidderRequest) {
+  const isInIframe = utils.inIframe();
+  let defaultParams;
+  defaultParams = {
+    ju: config.getConfig('pageUrl') || utils.getTopWindowUrl(),
+    jr: utils.getTopWindowReferrer(),
+    ch: document.charSet || document.characterSet,
+    res: `${screen.width}x${screen.height}x${screen.colorDepth}`,
+    ifr: isInIframe,
+    tz: new Date().getTimezoneOffset(),
+    tws: getViewportDimensions(isInIframe),
+    be: 1,
+    bc: bids[0].params.bc || `${BIDDER_CONFIG}_${BIDDER_VERSION}`,
+    auid: '540141567',
+    dddid: utils._map(bids, bid => bid.transactionId).join(','),
+    openrtb: '%7B%22mimes%22%3A%5B%22video%2Fmp4%22%5D%7D',
+    nocache: new Date().getTime()
+  };
+
+  if (utils.deepAccess(bidderRequest, 'gdprConsent')) {
+    let gdprConsentConfig = bidderRequest.gdprConsent;
+
+    if (gdprConsentConfig.consentString !== undefined) {
+      defaultParams.gdpr_consent = gdprConsentConfig.consentString;
+    }
+
+    if (gdprConsentConfig.gdprApplies !== undefined) {
+      defaultParams.gdpr = gdprConsentConfig.gdprApplies ? 1 : 0;
+    }
+
+    if (config.getConfig('consentManagement.cmpApi') === 'iab') {
+      defaultParams.x_gdpr_f = 1;
+    }
+  }
+
+  return defaultParams;
+}
+
+function buildOXBannerRequest(bids, bidderRequest) {
+  let queryParams = buildCommonQueryParamsFromBids(bids, bidderRequest);
+  queryParams.aus = utils._map(bids, bid => utils.parseSizesInput(bid.sizes).join(',')).join('|');
+
+  if (bids.some(bid => bid.params.doNotTrack)) {
+    queryParams.ns = 1;
+  }
+
+  if (bids.some(bid => bid.params.coppa)) {
+    queryParams.tfcd = 1;
+  }
+
+  let url = `https://${bids[0].params.delDomain}/v/1.0/avjp`
+  return {
+    method: 'GET',
+    url: url,
+    data: queryParams,
+    payload: {'bids': bids}
+  };
+}
+
+function handleVastResponse(response, serverResponse) {
+  const body = response.body
+  let bidResponses = [];
+  if (response !== undefined && body.vastUrl !== '' && body.pub_rev && body.pub_rev > 0) {
+    const openHtmltag = '<html><head><meta name="viewport" content="width=device-width, initial-scale=1.0"></head><body>';
+    const closeHtmlTag = '</body></html>';
+    const sdkScript = createSdkScript().outerHTML;
+    const placementDiv = createPlacementDiv();
+    placementDiv.dataset.pID = PUBLISHER_ID;
```

Comment:
has to be 'placementDiv.dataset.pId' with a capital 'i' lowercase 'd' in 'pId'

Reply:
great catch on this and lfid. updated both and the nit.